gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dcheckboxlabel_ttt2.lua

```diff
@@ -88,25 +96,71 @@ function PANEL:SetServerConVar(cvar)
 		end
 	end)
 
+	networkedVarTracker = networkedVarTracker % 1023 + 1
+	local myIdentifierString = "TTT2F1MenuServerConVarChangeCallback" .. tostring(networkedVarTracker)
+
 	local function OnServerConVarChangeCallback(conVarName, oldValue, newValue)
 		if not IsValid(self) then
-			cvars.RemoveChangeCallback(conVarName, "TTT2F1MenuServerConVarChangeCallback")
+			cvars.RemoveChangeCallback(conVarName, myIdentifierString)
 
 			return
 		end
 
 		self:SetValue(tobool(newValue), true)
 	end
 
-	cvars.AddChangeCallback(cvar, OnServerConVarChangeCallback, "TTT2F1MenuServerConVarChangeCallback")
+	cvars.AddChangeCallback(cvar, OnServerConVarChangeCallback, myIdentifierString)
+end
+
+---
+-- @param table databaseInfo containing {name, itemName, key}
+-- @realm client
+function PANEL:SetDatabase(databaseInfo)
+	if not istable(databaseInfo) then return end
+
+	local name = databaseInfo.name
+	local itemName = databaseInfo.itemName
+	local key = databaseInfo.key
+
+	if not name or not itemName or not key then return end
+
+	self.databaseInfo = databaseInfo
+
+	database.GetValue(name, itemName, key, function(databaseExists, value)
+		if databaseExists then
+			self:SetValue(value, true)
+		end
+	end)
+
+	self:SetDefaultValue(database.GetDefaultValue(name, itemName, key))
+
+	networkedVarTracker = networkedVarTracker % 1023 + 1
```

Comment:
Again here, why?

Also this is not a networked Variable -> naming wrong and if it is, because of some database module internals, i would rather see such logic inside the database module. So that it is not repeated that often

Reply:
First, I tried to explain above.
Second, yeah I can rename it, couldnt come up with a better name that combines the synced serverConvars and database Elements.

Reply:
I hope CallbackEnabledVar is more fitting. As that is the main infinte error loop we try to solve there.